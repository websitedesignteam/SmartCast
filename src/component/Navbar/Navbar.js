import React from 'react';
import styles from '../Navbar/Navbar.module.scss'
import { Link, matchPath, useLocation } from "react-router-dom"
function Navbar({openAuthModal, logoutUser, user}) {
	const location = useLocation();

	const isAuthActive = !!matchPath(
		location.pathname, 
		'/auth/:authType'
	  ); 

	const onClickLogin = () => {
		openAuthModal();
	}

	const onClickLogout = () => {
		logoutUser();
	}

  	return ( 
		<div className={styles.navbarContainer}>
			<Link className={styles.link} to='/'><strong>SmartCast</strong></Link>
			<div className={styles.linksContainer}>
				<Link className={styles.link} to='/'>Home</Link>
				{ !!user.access_token && <Link to='/profile' className={styles.link}>My Profile</Link>}
				{ (!user.access_token && !isAuthActive)
				? <button className={styles.userAction} onClick={onClickLogin}>Login</button>
				: !!isAuthActive 
				? <></>  
				: <button className={styles.userAction} onClick={onClickLogout}>Logout</button>
				}
			</div>
		</div>
  	);
}

export default Navbar;