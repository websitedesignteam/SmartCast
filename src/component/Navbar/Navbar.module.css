.navbarContainer{
       background-image: linear-gradient(to right, #895cf2 0%, #ffabf4 200%, #895cf2 100%);
       -webkit-box-shadow: 0px 3px 5px 0px rgba(0,0,0,0.37);
-moz-box-shadow: 0px 3px 5px 0px rgba(0,0,0,0.37);
box-shadow: 0px 3px 5px 0px rgba(0,0,0,0.37);
       display: flex;
       height: 60px;
       padding-left: 25px;
       padding-right: 25px;
}

.logoText{
       color: white;
       text-decoration: none;
}

.linksContainer{
       position: absolute;
       display: flex;
       flex-direction: row;
       right: 0;
       padding-left: 25px;
       padding-right: 25px;
       color: white;
}

.linkText{
       margin-left: 25px;
       margin-right: 25px;
}