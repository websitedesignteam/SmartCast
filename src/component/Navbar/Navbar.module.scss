@import "../../styles/variables";

.navbarContainer{
	background-image: linear-gradient(to right, #895cf2 0%, #ffabf4 200%, #895cf2 100%);
	-webkit-box-shadow: 0px 3px 5px 0px rgba(0,0,0,0.37);
	-moz-box-shadow: 0px 3px 5px 0px rgba(0,0,0,0.37);
	box-shadow: 0px 3px 5px 0px rgba(0,0,0,0.37);
	display: flex;
	height: 4rem;
	padding: 0 1.5rem;
	align-items: center;
	justify-content: space-between;
}

.link {
	text-decoration: none;
	font-size: 1.2rem;
	margin: 0 1rem 0 0;
}

.logoText{
	color: white;
	text-decoration: none;
}

.linksContainer{
	display: flex;
	flex-direction: row;
	color: white;
	margin: 0;
	.link:last-child {
		margin-right: 0;
	}
}

.userAction {
	background: none;
	font-size: 1.2rem;
}

.userAction:hover, .userAction:hover:not([disabled]) {
	background: none;
	outline: none;
}