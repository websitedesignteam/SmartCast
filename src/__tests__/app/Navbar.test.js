import React from 'react';
import { shallow } from 'enzyme';
import Navbar from 'src/component/Navbar/Navbar';

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => ({
      pathname: "localhost:3000/"
    })
}));

test("should render the Navbar", () => {
    const props = {
        user: {}
    }
    const component = shallow(<Navbar {...props} />);
    // expect(component.getElements()).toMatchSnapshot();
});