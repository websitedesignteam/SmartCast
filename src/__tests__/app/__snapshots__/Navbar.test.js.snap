// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Navbar 1`] = `
Array [
  <div
    className="navbarContainer"
  >
    <ForwardRef
      className="appLink"
      to="/"
    >
      <strong>
        SmartCast
      </strong>
      <img
        alt=""
        src="undefined/assets/logo.png"
      />
    </ForwardRef>
    <div
      className="linksContainer"
    >
      <ForwardRef
        className="link"
        title="Go to Home"
        to="/"
      >
        <img
          alt=""
          className="navIcon"
          src="undefined/assets/nav/home.svg"
        />
      </ForwardRef>
      <button
        className="userAction"
        onClick={[Function]}
        title="Login"
      >
        <img
          alt=""
          className="navIcon"
          src="undefined/assets/nav/login.svg"
        />
      </button>
    </div>
  </div>,
]
`;
